esi/src/parse.rs

```diff
@@ -249,26 +433,92 @@ fn parse_include<'a>(elem: &BytesStart) -> Result<Tag<'a>> {
     })
 }
 
+fn parse_assign<'a>(elem: &BytesStart) -> Result<Tag<'a>> {
+    let name = match elem
+        .attributes()
+        .flatten()
+        .find(|attr| attr.key.into_inner() == b"name")
+    {
+        Some(attr) => String::from_utf8(attr.value.to_vec()).unwrap(),
+        None => {
+            return Err(ExecutionError::MissingRequiredParameter(
+                String::from_utf8(elem.name().into_inner().to_vec()).unwrap(),
+                "name".to_string(),
+            ));
+        }
+    };
```

Comment:
This parsing pattern appears a few times in this file but, again, we can leave that for the parser rework.